pkg/fileservice/mem_cache.go

```diff
@@ -79,15 +87,15 @@ func (m *MemCache) Read(
 ) (
 	err error,
 ) {
-	_, span := trace.Start(ctx, "MemCache.Read")
-	defer span.End()
 
-	numHit := 0
+	var numHit, numRead int64
 	defer func() {
-		if m.stats != nil {
-			atomic.AddInt64(&m.stats.NumRead, int64(len(vector.Entries)))
-			atomic.AddInt64(&m.stats.NumHit, int64(numHit))
-		}
+		perfcounter.Update(ctx, func(c *perfcounter.Counter) {
```

Comment:
hmm, this call is still kind of clumsy.    anyway to simplify this call?

Reply:
We have to update counters in the context and counters passed in NewLocalFS and NewS3FS, and we have to specify which fields to increase.
Maybe we can use atomic.Int64 and the code will be c.CacheRead.Add(1).

Reply:
Changed to use atomic.Int64